src/Mvc/Mvc.Core/src/MvcOptions.cs

```diff
@@ -329,7 +329,20 @@ public int MaxModelBindingRecursionDepth
         /// Gets the <see cref="JsonSerializerOptions"/> used by <see cref="SystemTextJsonInputFormatter"/> and
         /// <see cref="SystemTextJsonOutputFormatter"/>.
         /// </summary>
-        public JsonSerializerOptions SerializerOptions { get; } = new JsonSerializerOptions();
+        public JsonSerializerOptions SerializerOptions { get; } = new JsonSerializerOptions
+        {
+            // Limit the object graph we'll consume to a fixed depth. This prevents stackoverflow exceptions
+            // from deserialization errors that might occur from deeply nested objects.
+            // This value is the same for model binding and Json.Net's serialization.
+            MaxDepth = DefaultMaxModelBindingRecursionDepth,
+
+            // Use camel casing for properties
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+
+            // Ignore casing when serializing values. Setting this to true keeps parity with Json.NET's
+            // behavior.
+            PropertyNameCaseInsensitive = true,
```

Comment:
How do you feel about this? In the absence of this property, it is required to send camel-case values to successfully deserialize. Most of our tests use pascal-case when sending data. 

We can update tests, but I'm not sure how sad users would be when random casing doesn't work.

Reply:
I think true is the right default for MVC. Also it keeps closer to Json.NET so less friction when upgrading. If someone wants to be strict then they can always change the setting.

Reply:
I just spoke to @rynowak offline and he's of the opinion that having it be case sensitive by default would be the right way to go. Users are always free to change it.

Reply:
We should use the best default regardless of migration. You can always turn this *ON* for migrating.

I'm interested to hear arguments about case-insensitive is the right default for a REST API.

Reply:
Made this case sensitive now and updated tests to be 🐫 cased

Reply:
> I'm interested to hear arguments about case-insensitive is the right default for a REST API.

Nothing other than it slightly reduces issues when using an API and I don't see much disadvantage. I don't feel strongly about it. As long as it is configurable

Reply:
We could always revisit this once we hear what users have to say with the preview5 bits.

Reply:
>  Nothing other than it slightly reduces issues when using an API and I don't see much disadvantage. I don't feel strongly about it. As long as it is configurable

APIs document their contract and are precise about what they support. JSON is a case sensitive format, and it's reasonable to follow along with that. I expect if we asked API users they would overwhelmingly support us following the spec.

Reply:
JSON being case sensitive in the browser is prior art, not a spec. There is no spec about how JSON should map to a .NET object :yum:

When it comes to JSON my philosophy has been be flexible in what you accept and exact in what you produce.